src/features/wallet/root/ActivityCard.tsx

```diff
@@ -34,51 +88,75 @@ const ActivityCard = forwardRef(
       },
     }))
 
-    const renderItem = ({ item, index }: { item: TxnData; index: number }) => (
-      <ActivityItem
-        type={item.type}
-        time={item.time}
-        amount={item.amount}
-        isFirst={index === 0}
-        isLast={index === data.length - 1}
-      />
-    )
+    const renderItem = ({
+      item,
+      index,
+    }: {
+      item: AnyTransaction | PendingTransaction
+      index: number
+    }) => {
+      return (
+        <ActivityItem
+          item={item}
+          isFirst={index === 0}
+          isLast={index === transactionData.length - 1}
+          address={address}
+        />
+      )
+    }
 
     const { dragMax, dragMid, dragMin } = animationPoints
 
+    const onFilterChanged = (f: FilterType) => {
+      LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut)
+      setFilter(f)
+      setTransactionData([])
+    }
+
+    useEffect(() => {
+      if (!transactionData.length) {
+        flatListRef?.current?.scrollToOffset({ animated: false, offset: 0 })
+      }
+    }, [transactionData.length])
+
     return (
       <BottomSheet
         ref={sheet}
         snapPoints={[dragMin, dragMid, dragMax]}
         initialSnapIndex={1}
         snapProgress={snapProgress}
-        renderHeader={() => <ActivityCardHeader />}
+        renderHeader={() => (
+          <ActivityCardHeader
+            filter={filter}
+            onFilterChanged={onFilterChanged}
+          />
+        )}
+        containerStyle={{ paddingHorizontal: m }}
         flatListProps={{
-          data,
-          keyExtractor: (item) => item.id,
+          style: { marginTop: n_m },
+          data: transactionData,
+          ref: flatListRef,
+          maxToRenderPerBatch: 50,
+          keyExtractor: (item: AnyTransaction | PendingTransaction) => {
+            if (isPendingTransaction(item)) {
+              return `${filter}.${(item as PendingTransaction).hash}`
+            }
+
+            return `${filter}.${(item as AddGatewayV1).hash}`
+          },
           renderItem,
+          refreshControl: (
+            <RefreshControl
+              refreshing={txnStatus === 'pending' && !transactionData.length}
```

Comment:
I dont see this when switching filters as it loads. I like the animation but I think it needs some kind of loading indicator or it feels like its frozen for a bit when loading transactions after changing filters.

Maybe we can add another boolean when the filter changes and its loading? We could also do this in a separate PR if you want as it works well already.

Reply:
hmmm. Are you testing on ios or android?

Reply:
The iOS simulator